doc/reference/networking/net_mgmt.rst

```diff
@@ -82,25 +86,35 @@ An example follows.
 	 * Set of events to handle.
 	 * See e.g. include/net/net_event.h for some NET_EVENT_xxx values.
 	 */
-	#define EVENT_SET (NET_EVENT_xxx | NET_EVENT_yyy)
+	#define EVENT_IFACE_SET (NET_EVENT_IF_xxx | NET_EVENT_IF_yyy)
+	#define EVENT_IPV4_SET (NET_EVENT_IPV4_xxx | NET_EVENT_IPV4_yyy)
 
-	struct net_mgmt_event_callback callback;
+	struct net_mgmt_event_callback iface_callback;
+	struct net_mgmt_event_callback ipv4_callback;
 
-	void callback_handler(struct net_mgmt_event_callback *cb, uint32_t mgmt_event,
+	void callback_handler(struct net_mgmt_event_callback *cb,
+			      uint32_t mgmt_event,
 			      struct net_if *iface)
 	{
-		if (mgmt_event == NET_EVENT_xxx) {
-			/* Handle NET_EVENT_xxx */
-		} else if (mgmt_event == NET_EVENT_yyy) {
-			/* Handle NET_EVENT_yyy */
+		if (mgmt_event == NET_EVENT_IF_xxx) {
+			/* Handle NET_EVENT_IF_xxx */
+		} else if (mgmt_event == NET_EVENT_IF_yyy) {
+			/* Handle NET_EVENT_IF_yyy */
+		} else if (mgmt_event == NET_EVENT_IPV4_xxx) {
+			/* Handle NET_EVENT_IPV4_xxx */
+		} else if (mgmt_event == NET_EVENT_IPV4_yyy) {
+			/* Handle NET_EVENT_IPV4_yyy */
 		} else {
 			/* Spurious (false positive) invocation. */
 		}
 	}
 
 	void register_cb(void)
 	{
-		net_mgmt_init_event_callback(&callback, callback_handler, EVENT_SET);
+		net_mgmt_init_event_callback(&iface_callback, callback_handler,
+					     EVENT_IFACE_SET);
+		net_mgmt_init_event_callback(&ipv4_callback, callback_handler,
+					     EVENT_IPV4_SET);
 		net_mgmt_add_event_callback(&callback);
```

Comment:
After a closer look at this, should this be:

```
net_mgmt_add_event_callback(&iface_callback);
net_mgmt_add_event_callback(&ipv4_callback);
```
?

Reply:
Indeed, good catch, will fix.